src/item.cpp

```diff
@@ -4742,6 +4742,36 @@ void item::mark_as_used_by_player(const player &p)
     used_by_ids += string_format( "%d;", p.getID() );
 }
 
+VisitResponse item::visit( const std::function<VisitResponse(item&)>& func ) {
+    switch( func( *this ) ) {
+        case VisitResponse::ABORT:
+            return VisitResponse::ABORT;
+
+        case VisitResponse::SKIP:
+            return VisitResponse::NEXT;
+
+        case VisitResponse::NEXT:
+            ; /* Continue below */
+    }
+
+    for( auto& e : contents ) {
+        auto r = func( e );
+
+        if( r == VisitResponse::NEXT ) {
+            r = e.visit( func );
```

Comment:
`e.visit` is this function and calls `func(*this)` again. In the end `func` is called twice for each content item.